lib/srv/discovery/discovery.go

```diff
@@ -1512,20 +1516,20 @@ func (s *Server) upsertDynamicMatchers(ctx context.Context, dc *discoveryconfig.
 	}
 	s.discardUnsupportedMatchers(&matchers)
 
-	awsServerFetchers, err := s.awsServerFetchersFromMatchers(s.ctx, matchers.AWS)
+	awsServerFetchers, err := s.awsServerFetchersFromMatchers(s.ctx, matchers.AWS, dc.GetName())
```

Comment:
Is there a cleaner way to do this than propagating discovery config name everywhere? For example, we already have relationship between discovery config and corresponding fetchers in the `dynamicServerXXXFetchers` maps - can this be used instead?

Reply:
The problem I was facing was that each Fetcher sends Instances using a channel.
On the other side, there's the SSMInstaller which receives Instances from every Fetcher. And so, at the SSMInstaller, we don't known which DiscoveryConfig a given instance is coming from.